asteroid_data_processing.py

```diff
@@ -0,0 +1,417 @@
+        params={
+            "format": "text",
+            "COMMAND": name,
+            "OBJ_DATA": "NO",
+            "EPHEM_TYPE": "OBSERVER",
+            "START_TIME": start_time,
+            "STOP_TIME": stop_time,
+            "STEP_SIZE": "1m",
+            "QUANTITIES": "1",
+        },
+    )
+    if response.status_code != 200:
+        logging.exception("Horizons API error")
+    asteroid_pos = separate_data(response, name, location)
+    return asteroid_pos
+
+
+@timeit
+def draw(
+    night_start,
+    night_end,
+    times,
+    altitudes,
+    b_times,
+    b_altitudes,
+    obs_start,
+    obs_end,
+    name,
+    min_deg,
+):
+    plt.plot(times, altitudes)
+    plt.plot(b_times, b_altitudes)
+    plt.xlabel("Time")
+    plt.ylabel("altitude (deg)")
+    plt.axhline(y=min_deg, color="green", linestyle="--", label="Min deg = 20°")
+
+    plt.axvline(
+        x=night_start,
+        color="blue",
+        linestyle="--",
+        label="Początek nocy astronomicznej",
+    )
+    plt.axvline(
+        x=night_end, color="blue", linestyle="--", label="Koniec nocy astronomicznej"
+    )
+    plt.title("A(t) " + name + " START =  " + str(obs_start) + "END = " + str(obs_end))
+    plt.ylim(0, 90)
+    plt.legend(loc="upper left")
+    date = datetime.today().date().strftime("%Y%m%d")
+    plt.savefig(f"graphs/{name}_altitude_{date}.png")
+    plt.clf()
+
+
+@timeit
+def get_stars_in_radius(radius, x_mean, y_mean):
+    coord = SkyCoord(ra=x_mean, dec=y_mean, unit=(u.deg, u.deg))
+    j = Gaia.cone_search(coord, radius * u.deg)
+    result = j.get_results()
+    return result
+
+
+@timeit
+def get_stars_in_d(MAX_DISTANCE, stars, a, b):
+    x = stars["ra"]
+    y = stars["dec"]
+    stars["d"] = abs((a * x + -1 * y + b) / np.sqrt(a**2 + 1))
+    stars = stars[stars["d"] < MAX_DISTANCE]
+    return stars
+
+
+@timeit
+def get_stars(radius, x_mean, y_mean, a, b, MAX_DISTANCE):
+    stars_in_radius = get_stars_in_radius(radius, x_mean, y_mean)
+    stars = get_stars_in_d(MAX_DISTANCE, stars_in_radius, a, b)
+    return stars
+
+
+@timeit
+def get_linear_f(x, y, x_mean, y_mean):
+    delta_x = x - x_mean
+    delta_y = y - y_mean
+
+    a = np.sum(delta_x * delta_y) / np.sum(delta_x**2)
+    b = y_mean - a * x_mean
+    return a, b
+
+
+def f(a, b, x):
+    return a * x + b
+
+
+def draw_f(ax, a, b, x):
+    ax.plot(x, f(a, b, x), color="red")
+    return ax
+
+
+def draw_circle(ax, radius, x_mean, y_mean):
+    center = (x_mean, y_mean)
+    circle = Circle(center, radius, edgecolor="black", facecolor="none")
+    ax.add_patch(circle)
+    xmin = center[0] - radius - 1
+    xmax = center[0] + radius + 1
+    ymin = center[1] - radius - 1
+    ymax = center[1] + radius + 1
+    ax.set_xlim(xmin, xmax)
+    ax.set_ylim(ymin, ymax)
+    return ax
+
+
+def draw_stars(ax, stars, size):
+    ax.plot(stars["ra"], stars["dec"], marker="*", ls="none", ms=size)
+    for i, mag in enumerate(stars["phot_g_mean_mag"]):
+        if mag > 18:
```

Comment:
such "magic numbers" should typically be moved to the top of the file

```suggestion
        if mag > MAG_MAX:
```